src/game/engine/internals/Rigidbody.js

```diff
@@ -0,0 +1,46 @@
+import { Vector } from '../../shared';
+
+const GRAVITY_ACCELERATION = 0.001;
+
+class Rigidbody {
+  #resultVector = Vector.Zero;
+
+  #velocityVector = Vector.Zero;
+
+  #accelerationVector = Vector.Zero;
+
+  addGravity() {
+    this.addAcceleration(new Vector(0, GRAVITY_ACCELERATION));
+  }
+
+  /**
+   * @param {import('../../shared').Vector} v
+   */
+  addVelocity(v) {
+    this.#velocityVector = v;
+  }
+
+  /**
+   * @param {import('../../shared').Vector} v
+   */
+  addAcceleration(v) {
+    this.#accelerationVector = this.#accelerationVector.add(v);
+    this.#resultVector = this.#resultVector.add(this.#accelerationVector);
+  }
+
+  #findResultVector() {
+    this.#velocityVector = this.#velocityVector.add(this.#accelerationVector);
+    this.#resultVector = this.#resultVector.add(this.#velocityVector);
```

Comment:
Wydaje mi się, że `#resultVector` nie jest nam w ogóle potrzebny, bo wektorem przesunięcia w danym kroku będzie tak naprawdę `#velocityVector`: żeby otrzymać nową pozycję `transform` wystarczy, że w każdym `frame` do bieżącej pozycji `transform.origin` doda się wyliczony `#velocityVector`:
```js
this.#velocityVector = this.#velocityVector.add(this.#accelerationVector);
```

Reply:
Wydaje mi się, że sposobem, który pokazałeś wyżej, dodajemy do `velocityVector` sam tylko wektor przyspieszenia, ale trzeba jeszcze znów dodać wektor prędkości, który zmienia się tylko o to przyspieszenie. Gdybym nie użyła `resultVector`, tylko zwiększyła `velocityVector` o samego siebie, to w kolejnym update `velocityVector` będzie za duży.

Reply:
Ja się osobiście zgadzam z Hubertem. Całość logiki update to (w pseudokodzie):
```
przyspieszenie = // liczone z sił, bo a = F / m
prędkość = poprzednia_prędkość + przyspieszenie // bo v = a * t
położenie = poprzednie_położenie + prędkość // bo x = v * t
```
(gdzie dla uproszczenia t = 1, m = 1)

Więc na nasze potrzeby:
- przyspieszenie <=> `this.#accelerationVector`
- prędkość <=> `this.#velocityVector`
- położenie <=> `transform.origin`

I nigdzie nie widać potrzeby dla `resultVector`.

Reply:
Okej, już widzę problem, jest w `Bird.js` 😂